### PR TITLE
Cleanup storehaus-redis

### PR DESCRIPTION
**Pull request by BenFradet**

### CODE REVIEW

**Comment by BenFradet** [comment]
This will be made better once #310 is done.

File: storehaus-redis/src/main/scala/com/twitter/storehaus/redis/RedisHashStore.scala [hunk 1]
```diff
@@ -45,7 +44,7 @@ class RedisHashStore(val client: Client, ttl: Option[Duration])
   extends Store[ChannelBuffer, Map[ChannelBuffer, ChannelBuffer]] {
 
   override def get(k: ChannelBuffer): Future[Option[Map[ChannelBuffer, ChannelBuffer]]] =
-    client.hGetAll(k).map({ case e if (e.isEmpty) => None case xs => Some(Map(xs:_*)) })
+    client.hGetAll(k).map { case e if e.isEmpty => None case xs => Some(Map(xs: _*)) }
```
Comment (johnynek) [inline storehaus-redis/src/main/scala/com/twitter/storehaus/redis/RedisHashStore.scala]: can we make two lines:

``` scala
.map {
  case e if e.isEmpty => None
  case xs => Some(Map(xs: _*))
}
```

**Review by johnynek — CHANGES_REQUESTED** [review]

File: storehaus-redis/src/main/scala/com/twitter/storehaus/redis/RedisHashStore.scala (hunk 1 above)
Comment (BenFradet) [reply storehaus-redis/src/main/scala/com/twitter/storehaus/redis/RedisHashStore.scala]: done :+1:

File: storehaus-redis/src/main/scala/com/twitter/storehaus/redis/RedisSortedSetStore.scala [hunk 2]
```diff
@@ -34,7 +33,8 @@ object RedisSortedSetStore {
  */
 class RedisSortedSetStore(client: Client)
   extends MergeableStore[ChannelBuffer, Seq[(ChannelBuffer, Double)]] {
-  def semigroup = implicitly[Semigroup[Seq[(ChannelBuffer, Double)]]]
+  def semigroup: Semigroup[Seq[(ChannelBuffer, Double)]] =
+    implicitly[Semigroup[Seq[(ChannelBuffer, Double)]]]
```
Comment (johnynek) [inline storehaus-redis/src/main/scala/com/twitter/storehaus/redis/RedisSortedSetStore.scala:37]: Is this really the semigroup that redis is doing? I tend to doubt it. Seems like the value is more like `Map[ChannelBuffer, Double]`. Can we add a test that the semigroup we declare here is consistent with how redis works?

We could do that by comparing `.multiMerge` here on a big set of items to what we would get from `MergeableStore.fromStore` and using the semigroup here.

**Comment by BenFradet** [comment]
ping @johnynek :smile:

**Review by johnynek — COMMENTED** [review]
sorry, I had a comment, but somehow it didn't get submitted. Maybe we don't have to fix it now, but we should file an issue, or maybe you can correct my understanding.

**Pull request merged**